### PR TITLE
[15.0][MIG] pricelist_cache

### PR DESCRIPTION
**Pull request by juanpgarza**
Standard migration

### CODE REVIEW

**Review by rousseldenis — CHANGES_REQUESTED** [review]
@juanpgarza Could you fix tests ?

**Comment by juanpgarza** [comment]
hi @rousseldenis 
I would but I don't know how. 
Could you help me a little or point me to a tutorial or webinar on the subject?

**Comment by valentincastravete** [comment]
Hi @juanpgarza 
Could you re-run the tests so I can lend you a hand please?
Their logs have been deleted.
Thanks!

**Comment by valentincastravete** [comment]
> Hi @juanpgarza Could you re-run the tests so I can lend you a hand please? Their logs have been deleted. Thanks!

Hi @rousseldenis Could you re-run the tests please? Thanks!

File: pricelist_cache/models/product_pricelist.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        "product.pricelist",
+        relation="product_pricelist_cache__parent_pricelist_ids_rel",
+        column1="pricelist_id",
+        column2="parent_pricelist_id",
+        compute="_compute_parent_pricelist_ids",
+        store=True,
+    )
+    is_pricelist_cache_computed = fields.Boolean()
+    is_pricelist_cache_available = fields.Boolean(
+        compute="_compute_is_pricelist_cache_available", store=True
```
Comment (valentincastravete) [inline pricelist_cache/models/product_pricelist.py:22]:
```suggestion
        compute="_compute_is_pricelist_cache_available",
        store=True,
        recursive=True,
```
_/opt/odoo/odoo/fields.py:746: UserWarning: Field product.pricelist.is_pricelist_cache_available should be declared with recursive=True_

**Comment by valentincastravete** [comment]
Hi @juanpgarza 
Could you please update your repository with the last updates on the pricelist_cache module on v14 please?
To see it changes anything related to the tests.
Thanks!

File: pricelist_cache/tests/common.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        {"id": 19, "price": 100.0},
+        {"id": 20, "price": 47.0},
+    ],
+    "pricelist_cache.list5": [
+        {"id": 17, "price": 45.0},
+        {"id": 18, "price": 99.0},
+        {"id": 19, "price": 120.0},
+        {"id": 20, "price": 67.0},
+    ],
+}
```
Comment (valentincastravete) [inline pricelist_cache/tests/common.py:45]:
```suggestion
LIST_PRICES_MAPPING = {
    "pricelist_cache.list0": [
        {"id": 17, "price": 100.0},
        {"id": 18, "price": 15.8},
        {"id": 19, "price": 100.0},
        {"id": 20, "price": 47.0},
    ],
    "pricelist_cache.list1": [
        {"id": 17, "price": 75.0},
        {"id": 18, "price": 15.8},
        {"id": 19, "price": 100.0},
        {"id": 20, "price": 47.0},
    ],
    "pricelist_cache.list2": [
        {"id": 17, "price": 50.0},
        {"id": 18, "price": 15.8},
        {"id": 19, "price": 100.0},
        {"id": 20, "price": 47.0},
    ],
    "pricelist_cache.list3": [
        {"id": 17, "price": 25.0},
        {"id": 18, "price": 79.0},
        {"id": 19, "price": 100.0},
        {"id": 20, "price": 47.0},
    ],
    "pricelist_cache.list4": [
        {"id": 17, "price": 15.0},
        {"id": 18, "price": 50.0},
        {"id": 19, "price": 100.0},
        {"id": 20, "price": 47.0},
    ],
    "pricelist_cache.list5": [
        {"id": 17, "price": 45.0},
        {"id": 18, "price": 35.8},
        {"id": 19, "price": 120.0},
        {"id": 20, "price": 67.0},
    ],
}
```

Comment (valentincastravete) [reply pricelist_cache/tests/common.py:45]: Hi @juanpgarza 
I've found the solution to the tests.
It's caused by the odoo/addons/product module demo data change of a product:

The one used here:
https://github.com/juanpgarza/sale-workflow/blob/15.0-mig-pricelist_cache/pricelist_cache/data/demo.xml#L121
With the ID 18;
image


V14.0:
https://github.com/odoo/odoo/blob/14.0/addons/product/data/product_demo.xml#L285
Price: 79.0

V15.0:
https://github.com/odoo/odoo/blob/15.0/addons/product/data/product_demo.xml#L285
Price: 15.8


After updating the price, the tests passes.

Regards

**Review by valentincastravete — CHANGES_REQUESTED** [review]

File: pricelist_cache/tests/common.py (hunk 2 above)
Comment (valentincastravete) [reply pricelist_cache/tests/common.py:45]: @rousseldenis Could you apply this suggestion for him please? So it can be merged

**Comment by valentincastravete** [comment]
@rousseldenis Hi, my PR has been merged, this can be closed.

**Pull request closed without merging**